flink-table/flink-sql-gateway-api/src/main/java/org/apache/flink/table/gateway/api/results/FunctionInfo.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.gateway.api.results;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.table.functions.FunctionDefinition;
+import org.apache.flink.table.functions.FunctionIdentifier;
+import org.apache.flink.table.functions.FunctionKind;
+
+import javax.annotation.Nullable;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Info to describe the function. It is not equivalent to the {@link FunctionDefinition} that needs
+ * to load the implementation, which may require to download the jar from the remote to the local
+ * machine and load the classes. Comparing to the {@link FunctionDefinition}, the {@link
+ * FunctionInfo} return the available information in the current state, which is much lighter.
+ */
+@PublicEvolving
+public class FunctionInfo {
+
+    /** Identifier of the function. */
+    private final FunctionIdentifier identifier;
+    /** Kind of the function. If the value is null, it means kind of the function is unresolved. */
+    private final @Nullable FunctionKind kind;
+
```

Comment:
Did FunctionKind has member like 'UNKOWN'? If did, I think it's better to use it instead of 'null'.

Reply:
FunctionKind has members OTHERS. I think OTHERS is much different from the UNKNOWN. UNKNOWN means we don't resolve this but the OTHERS means we have already resolved the Function but the result type doesn't belong to TABLE/SCALAR or so.